### PR TITLE
User Management - Issue #717

### PR DESCRIPTION
**Pull request by zawapete**
## Description

See #717 

## Related Issue

See #717 

## Motivation and Context

See #717 

## How Has This Been Tested?

Manually
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: client/src/javascript/components/AppWrapper.js [hunk 1]
```diff
@@ -182,10 +183,11 @@ class AuthEnforcer extends React.Component {
     }
 
     if (this.state.isAuthenticated && !this.state.isClientConnected) {
+      const isInitialUser = this.state.isInitialUser;
```
Comment (jfurrow) [inline client/src/javascript/components/AppWrapper.js]:
```js
const {isInitialUser} = this.state;
```

File: client/src/javascript/components/general/ClientConnectionInterruption.js [hunk 2]
```diff
@@ -109,8 +109,30 @@ export default class ClientConnectionInterruption extends React.Component {
   }
 
   render() {
+    const isAdmin = AuthStore.isAdmin();
+    const isInitialUser = this.props.isInitialUser;
```
Comment (jfurrow) [inline client/src/javascript/components/general/ClientConnectionInterruption.js]:
```js
const {isInitialUser} = this.props;
```

**Comment by zawapete** [comment]
@jfurrow done

File: server/routes/auth.js [hunk 3]
```diff
@@ -139,6 +139,12 @@ router.get('/logout', (req, res) => {
   res.clearCookie('jwt').send();
 });
 
+router.use('/users', (req, res, next) => {
+  if (req.user && req.user.isAdmin) {
+    next();
+  }
```
Comment (jfurrow) [inline server/routes/auth.js]: We should respond with status code `403`. I didn't test this but this should work:

```js
router.use('/users', (req, res, next) => {
  if (req.user && req.user.isAdmin) {
    next();
    return;
  }

  res.status(401).send('Must be admin');
});
```

**Pull request merged**